analyzers/src/SonarAnalyzer.Common/Rules/MultipleVariableDeclarationBase.cs

```diff
@@ -18,50 +18,40 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-using System.Collections.Generic;
-using System.Linq;
-using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.Diagnostics;
-using SonarAnalyzer.Helpers;
-
 namespace SonarAnalyzer.Rules
 {
-    public abstract class MultipleVariableDeclarationBase : SonarDiagnosticAnalyzer
+    public struct MultipleVariableDeclarationConstants
     {
         internal const string DiagnosticId = "S1659";
-        protected const string MessageFormat = "Declare '{0}' in a separate statement.";
-
-        protected abstract GeneratedCodeRecognizer GeneratedCodeRecognizer { get; }
     }
 
-    public abstract class MultipleVariableDeclarationBase<TLanguageKindEnum,
-        TFieldDeclarationSyntax, TLocalDeclarationSyntax> : MultipleVariableDeclarationBase
-        where TLanguageKindEnum : struct
-        where TFieldDeclarationSyntax: SyntaxNode
-        where TLocalDeclarationSyntax: SyntaxNode
+    public abstract class MultipleVariableDeclarationBase<TSyntaxKind> : SonarDiagnosticAnalyzer<TSyntaxKind>
+        where TSyntaxKind : struct
     {
+        protected override string MessageFormat => "Declare '{0}' in a separate statement.";
+
+        protected MultipleVariableDeclarationBase() : base(MultipleVariableDeclarationConstants.DiagnosticId) { }
+
         protected sealed override void Initialize(SonarAnalysisContext context)
         {
             context.RegisterSyntaxNodeActionInNonGenerated(
-                GeneratedCodeRecognizer,
+                Language.GeneratedCodeRecognizer,
                 c =>
                 {
-                    var local = (TLocalDeclarationSyntax)c.Node;
-                    CheckAndReportVariables(GetIdentifiers(local).ToList(), c, SupportedDiagnostics[0]);
+                    CheckAndReportVariables(Language.Syntax.LocalDeclarationIdentifiers(c.Node), c, SupportedDiagnostics[0]);
```

Comment:
This historical workaround of `SupportedDiagnostics[0]` used to be a workaround, now it's promoted to a bad practice 😆 
```suggestion
                    CheckAndReportVariables(Language.Syntax.LocalDeclarationIdentifiers(c.Node), c, Rule);
```